tests/unit_tests/test_game.py

```diff
@@ -91,6 +91,12 @@ def test_add_game_connection_throws_if_not_lobby_state(game: Game, players, mock
 
     game.on_game_end.assert_any_call()
 
+async def test_game_marked_dirty_when_timed_out(game: Game):
```

Comment:
So, testing this behavior requires actually making it not sleep inside of `timeout_game`, otherwise our test suite is going to be unnecessarily delayed. You can do this by mocking out `asyncio.sleep()` to be a null coroutine that immediately returns.

See the [`@patch` decorator](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch)